pydrive2/fs/spec.py

```diff
@@ -542,6 +542,34 @@ def cp_file(self, lpath, rpath, **kwargs):
             buffer = io.BytesIO(stream.read())
             self.upload_fobj(buffer, rpath)
 
+    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+
+        if recursive:
+            raise NotImplementedError("Recursive move is not supported")
+        if maxdepth:
+            raise NotImplementedError("Max depth move is not supported")
+
+        file1_name = posixpath.basename(path1)
+        file2_name = posixpath.basename(path2)
+
+        file1_parent = self._parent(path1)
+        file2_parent = self._parent(path2)
+
+        file1_id = self._get_item_id(path1)
+
+        file1 = self.client.CreateFile({"id": file1_id})
+
+        file1.FetchMetadata(fields="title,parents")
+
+        if file1_name != file2_name:
+            file1["title"] = file2_name
+
+        if file1_parent != file2_parent:
+            file2_parent_id = self._get_item_id(file2_parent)
```

Comment:
Q / check: can path2 be an absolute one, or is it still within the self.path directory.

Do we want to support a way to move files outside, support absolute paths?

Reply:
check what happens if `file2_parent` is empty

Reply:
> Q / check: can path2 be an absolute one, or is it still within the self.path directory.
> Do we want to support a way to move files outside, support absolute paths?

```python
fs = GDriveFileSystem("root/tmp", auth)
fs.mv('root/tmp/a.pdf', 'root/test/b.pdf')
```

This worked. so we can move files even outside `self.path`.



> check what happens if file2_parent is empty

`file2_parent` cannot be empty otherwise `file2_parent_id = self._get_item_id(file2_parent)` will fail.
but i tried to force:
```python
file2_parent_id = None # or ""
file1["parents"] = [{"id": file2_parent_id}]
```
and i got an error:

```python
Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 780, in _FilesUpdate
    self.auth.service.files()
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/http.py", line 938, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 404 when requesting https://www.googleapis.com/drive/v2/files/1KSvXEOjUdDwTR2EfKm6Zl2xKX1OEmJgY?supportsAllDrives=true&alt=json returned "File not found:". Details: "[{'domain': 'global', 'reason': 'notFound', 'message': 'File not found: ', 'locationType': 'other', 'location': 'file'}]">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/tests/gdrivefs.py", line 86, in <module>
    fs.mv('root/tmp/b.pdf', 'root/tmp/aa/a.pdf')
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/decorators.py", line 45, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/flow.py", line 127, in retry
    return call()
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/decorators.py", line 66, in __call__
    return self._func(*self._args, **self._kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/fs/spec.py", line 572, in mv
    file1._FilesUpdate()
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/auth.py", line 84, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 56, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 785, in _FilesUpdate
    raise ApiRequestError(error)
pydrive2.files.ApiRequestError: <HttpError 404 when requesting https://www.googleapis.com/drive/v2/files/1KSvXEOjUdDwTR2EfKm6Zl2xKX1OEmJgY?supportsAllDrives=true&alt=json returned "File not found:". Details: "[{'domain': 'global', 'reason': 'notFound', 'message': 'File not found: ', 'locationType': 'other', 'location': 'file'}]">
```

so, `file2_parent_id` needs to be a valid id.

Reply:
Q: so, what is the usual interface for the other existing calls (e.g. remove, or ls, etc) - does it expect a relative path (w/o `root`) or an absolute one with `root`?

Reply:
I'm not sure, but i think everything want an absolute path.

fot example `ls` return `None` or get buggy with relative paths.

es:

```python
fs = GDriveFileSystem("root/tmp", auth)

# self.path
print(fs.ls("root/tmp"))
# > ['root/tmp/folder1/', [other files]]
print(fs.ls("root/tmp/folder1"))
# > ['root/tmp/folder1/folder2/', 'root/tmp/folder1/file1']
print(fs.ls("root/tmp/folder1/folder2"))
# > ['root/tmp/folder1/folder2/file2']

# self.path = root/tmp
# .
# |- folder1
# |  |- file1
# |  |- folder2
# |  |  |- file2

print(fs.ls("."))
# > ['./tmp/', './[files in my drive root]/']
print(fs.ls("./folder1"))
# > None
print(fs.ls("./folder1/folder2"))
# > None
print(fs.ls("folder1"))
# > ['folder1/tmp/', 'folder1/[files in my drive root]/']
print(fs.ls("folder1/folder2"))
# > None
```